lint/util.py

```diff
@@ -610,3 +586,45 @@ def load_json(*segments, from_sl_dir=False):
     base_path = "Packages/SublimeLinter/" if from_sl_dir else ""
     full_path = base_path + "/".join(segments)
     return sublime.decode_value(sublime.load_resource(full_path))
+
+
+# syntax functions
+
+
+def yield_tail_scopes(scope):
+    while "." in scope:
+        scope, tail = scope.rsplit(".", 1)
+        yield tail
+
+
+def max_selector_score(view, selectors):
+    return max((view.score_selector(0, s) for s in selectors if s) or [0])
+
+
+def get_syntax_by_score(view, syntax_map):
+    try:
+        scoring = {
+            max_selector_score(view, selectors): syntax
+            for syntax, selectors in syntax_map.items() if selectors
+        }
+    except ValueError:
+        return
+    else:
+        max_score = max(scoring)
+        if max_score > 0:
+            return scoring[max_score]
+
+
+def get_syntax(view):
+    """Return the view's syntax."""
+    syntax_map = load_json('resources/syntax_map.json', from_sl_dir=True)
+    main_scope = view.scope_name(0).split()[0]
+
+    # lazy syntax lookup
+    for tail in yield_tail_scopes(main_scope):
+        if tail in syntax_map:
+            return tail
+
+    # more elaborate lookup via scoring mechanism
+    # or last tail as fallback, e.g. `python` from `source.python`
+    return get_syntax_by_score(view, syntax_map) or tail
```

Comment:
`tail` will be `'source'` or `'text'` here and not what we want.

Reply:
No. I will not. ;-)